src/index.test.ts

```diff
@@ -69,3 +70,42 @@ test('it should verify a token', async () => {
     'iss': 'token.dev',
   })
 })
+
+test('it should sign and verify a token', async () => {
+  let generator = new Generator({
+    key: 'test-rsa',
+    issuer: 'token.dev',
+    audience: 'foo',
+  })
+
+  let token = generator.sign({
+    foo: 'bar',
+  }, 1000, 'test')
+
+  let result = generator.verify(token) as JwtPayload & { foo: string }
```

Comment:
Quiero que podamos evitar este casting haciéndolo dentro.

```suggestion
  let result = generator.verify<XXX>(token)
```